include/pybind11/stl_bind.h

```diff
@@ -115,6 +115,14 @@ void vector_modifiers(enable_if_t<is_copy_constructible<typename Vector::value_t
     using SizeType = typename Vector::size_type;
     using DiffType = typename Vector::difference_type;
 
+    auto wrap_i = [](DiffType i, SizeType n) {
+        if (i < 0)
+            i += n;
+        if (i < 0)
```

Comment:
Wouldn't it be better to make this check `if (i < 0 || i > n)` and then remove the exceptions elsewhere?